pkg/controller/unifiedpushserver/unifiedpushserver_controller.go

```diff
@@ -187,6 +187,26 @@ func (r *ReconcileUnifiedPushServer) Reconcile(request reconcile.Request) (recon
 		}
 	} else if err != nil {
 		return reconcile.Result{}, err
+	} else {
+		desiredImage := postgresql.image()
+
+		containerSpec := findContainerSpec(foundPostgresqlDeployment, POSTGRES_CONTAINER_NAME)
+		if containerSpec == nil {
```

Comment:
Are there situations where this could happen? :thinking: If so, should we stop progress and requeue or something, rather than just writing an info-level log message?

Reply:
I actually didn't know what to do and simply logged a message :grin: 

> Are there situations where this could happen?

Yeah, when somebody changes the container name manually in the deployment. Or, if there's a mismatch in those among different operator versions.

> If so, should we stop progress and requeue or something

What do you mean by this?

Reply:
Why I mean is that if we enter this block, something has gone terribly wrong that we should deal with, but I'm not 100% sure what to do.

Since the operator is the owner of the deployment, I think it should reconcile to its own state of the world -- if someone else changed something manually, it should change it back. If an earlier version of the operator did something differently, this version should update to its own way (similar to how this PR is making sure it gets updated to the expected images, I think eventually it should update to the expected _everything_).

What I think I'm saying is that I think we should be doing something equivalent to `kubectl apply`, rather than checking for specific fields, and allowing for variation in everything else. Does that make sense? I'm not sure if it's possible, or not. Maybe doing update with the `postgresqlDeployment` object rather than hoping that the `foundPostgresqlDeployment` object has the right structure for us to be able to update the image fields _might_ be possible (I haven't tried).

Again though, this PR works, so is good progress and could be merged without trying to go down that rabbit hole! :smile:

Reply:
Then instead of checking specific keys in the deployment, we should compare everything.

@grdryn I think what you say makes sense. However, I don't see anything like that in other operators in RHMI. So, not a best practice perhaps. But I cannot tell why.

ping @wei-lee @maleck13
(summary: this PR ensures the image names and versions in the deployment. my code only checks the image field but @grdryn asks why we should not simply compare everything.)

Reply:
I think for now just checking the image field is enough, that is the main concern for us for the moment. We can improve this on later on depending on what we see how people using this operator.

Reply:
@grdryn gonna rebase and merge this one then

Reply:
Sounds good! :+1: